### PR TITLE
Scan: Implement reinvest

### LINKED ISSUE #2770: Implement Reinvest functionality
Add the reinvest button on the withdraw reward button's right side.

Note: 
Reinvest means sending the withdraw reward and delegate messages on the same transaction.

### PR DESCRIPTION
**Pull request by Pitchanai**
Fixed: #2770 

## Implementation details
- Implement reinvest

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: scan/src/components/modal/ValidatorStakingInfo.re [hunk 1]
```diff
@@ -143,6 +143,20 @@ module StakingInfo = {
...
+            SubmitMsg.Reinvest(
+              validatorAddress,
+              switch (balanceAtStakeSub) {
+              | Data({reward: {amount}}) => amount
+              | _ => 0.
+              },
+            ),
+          ),
+        ),
+      );
```
Comment (pzshine) [inline scan/src/components/modal/ValidatorStakingInfo.re]: Could you change the code format to

```
 let reinvest = () =>
    (
      validatorAddress,
      switch (balanceAtStakeSub) {
      | Data({reward: {amount}}) => amount
      | _ => 0.
      },
    )
    ->SubmitMsg.Reinvest
    ->SubmitTx
    ->OpenModal
    ->dispatchModal;



```

**Review by pzshine — CHANGES_REQUESTED** [review]

File: CHANGELOG_UNRELEASED.md [hunk 2]
```diff
@@ -32,6 +32,7 @@
 
 ### Scan
 
+- (impv) [\#2854](https://github.com/bandprotocol/bandchain/pull/2854) Implement reinvest
```
Comment (evilpeach) [inline CHANGELOG_UNRELEASED.md]: feat

File: scan/src/components/modal/ValidatorStakingInfo.re [hunk 3]
```diff
@@ -199,18 +209,33 @@ module StakingInfo = {
...
+                py=5
+                onClick={_ => reinvest()}
+                disabled={
+                  switch (allSub) {
+                  | Data((_, balanceAtStake, _)) => balanceAtStake.reward.amount <= 0.
+                  | _ => true
+                  }
+                }>
+                <Text value="Reinvest" weight=Text.Medium nowrap=true block=true />
+              </Button>
```
Comment (evilpeach) [inline scan/src/components/modal/ValidatorStakingInfo.re]: 1. Let's move switch on allSub above the `<div>` like this.
2. Pass the amount to reinvest function.
```
{switch(allSub) {
     | Data((_, balanceAtStake, _)) => {
          let disable = balanceAtStake.reward.amount <= 0.;
          <div className={CssHelper.flexBox()} id="withdrawRewardContainer">
               // render 2 button here and use disable variable.
          </div>
     }
     | _ => 
           <div className={CssHelper.flexBox()}>
               // render 2 button here and set disable as true
          </div>
     }
}}
```

File: scan/src/components/modal/ValidatorStakingInfo.re [hunk 4]
```diff
@@ -140,9 +139,22 @@ module StakingInfo = {
 
     let allSub = Sub.all3(infoSub, balanceAtStakeSub, unbondingSub);
 
-    let withdrawReward = () =>
-      dispatchModal(OpenModal(SubmitTx(SubmitMsg.WithdrawReward(validatorAddress))));
+    let withdrawReward = () => {
+      validatorAddress->SubmitMsg.WithdrawReward->SubmitTx->OpenModal->dispatchModal;
+    };
 
+    let reinvest = () =>
```
Comment (evilpeach) [inline scan/src/components/modal/ValidatorStakingInfo.re]: pass amount as parameter

File: scan/src/components/modal/submitTx/ReinvestMsg.re [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  let validatorInfoSub = ValidatorSub.get(validator);
+
+  React.useEffect0(_ => {
+    let msgsOpt = {
+      let%Opt amountValue = Some(amount);
+      Some([|
+        TxCreator.WithdrawReward(validator),
+        TxCreator.Delegate(
+          validator,
+          {amount: amountValue |> Js.Float.toFixedWithPrecision(~digits=0), denom: "uband"},
```
Comment (evilpeach) [inline scan/src/components/modal/submitTx/ReinvestMsg.re]: I guess this function will automatically round up if the remainder is more than 5. So, I think you have to use `amountValue->Js.Math.floor_float->Js.Float.toString` to do it.

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by pzshine — APPROVED** [review]

**Pull request merged**